grudge/symbolic/operators.py

```diff
@@ -651,20 +654,32 @@ def matrix(self, afgrp, volgrp, dtype):
                     afgrp.nunit_dofs),
                 dtype=dtype)
 
-        from modepy.tools import UNIT_VERTICES
         import modepy as mp
+
+        vol_basis = volgrp.basis_obj()
+        faces = mp.faces_for_shape(volgrp.shape)
+
+        # NOTE: Assumes each face has the same quadrature/basis
```

Comment:
Why not move this computation inside the face loop, below? You can zip that with the `face_vertex_indices` loop, or better yet, get the face vertex indices via the shape. `face_vertex_indices` in meshmode should probably be deprecated in favor of using more modepy (feel free to do here or file an issue).

Reply:
Wasn't sure if it would be potentially expensive to keep recomputing this for each face, but I guess it's okay?

Also BTW we don't need `face_vertex_indices` at all since each `face` knows how to associate the unit nodes (on the volume element) and itself